File: DFS/dfs.c
//Time complexity O(V+E)
#include<stdio.h>
#include<stdlib.h>
void DFS(int **graph, int *visited, int n, int i)
{
    int j;
	printf("\n%d",i);
    visited[i]=1;
	
	for(j=0;j<n;j++)
       if(!visited[j]&&graph[i][j]==1)
            DFS(graph,visited,n,j);
}

 
void main()
{
    int i,j,n;
    printf("Enter number of vertices:");
	scanf("%d",&n);
	int **graph = (int**)malloc(n*sizeof(int *));	//graph in the form of adjency matrix
 	for(i=1;i<=n;i++){
 	graph[i]=(int *) malloc(n*sizeof(int));
 	}
 	 int *visited=(int *)malloc(n*sizeof(int));	//visited array
    //read the adjecency matrix
	printf("\nEnter adjecency matrix of the graph:");
   
	for(i=0;i<n;i++)
       for(j=0;j<n;j++)
			scanf("%d",&graph[i][j]);
 
    //visited is initialized to zero
   for(i=0;i<n;i++)
        visited[i]=0;
 
    DFS(graph,visited,n,0);
}
 

File: DFS/dfs.java
import java.io.*;
import java.util.*;

public class dfs {
    public static void Dfs(int graph[][],int visited[],int n,int i){
        int j;
	System.out.println(i);
    visited[i]=1;
	
	for(j=0;j<n;j++)
       if((visited[j])==0 && graph[i][j]==1)
            Dfs(graph,visited,n,j);

    }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        System.out.println("enter the number of vertices:");
        int n=sc.nextInt();
        int[][] graph=new int[n][n];
        int[] visited=new int[n];
        System.out.println("enter the adjency matrix");
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                graph[i][j]=sc.nextInt();
            }
        }
        for(int i=0;i<n;i++){
            visited[i]=0;
        }
    Dfs(graph,visited,n,0);
    }
}